src/main/kotlin/uk/gov/justice/digital/hmpps/assessrisksandneeds/services/OnnxPredictorService.kt

```diff
@@ -0,0 +1,236 @@
+package uk.gov.justice.digital.hmpps.assessrisksandneeds.services
+
+import ai.onnxruntime.OnnxTensor
+import ai.onnxruntime.OrtEnvironment
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.OffenderAndOffencesDto
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.PredictorSubType
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.PredictorType
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.RiskPredictorsDto
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.Score
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.api.model.ScoreType
+import uk.gov.justice.digital.hmpps.assessrisksandneeds.services.exceptions.ONNXResponseFailure
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.Period
+import java.time.format.DateTimeFormatter
+
+@Service
+class OnnxPredictorService(@Value("\${onnx-predictors.onnx-path}") private val onnxFilePath: String) {
+
+  fun calculatePredictorScores(
+    offenderAndOffences: OffenderAndOffencesDto,
+  ): RiskPredictorsDto {
+
+    log.info("Generating RSR score using ONNX runtime for CRN ${offenderAndOffences.crn}")
+    val results = OrtEnvironment.getEnvironment().use { env ->
+      env.createSession(onnxFilePath).use { session ->
+
+        log.info("Loaded ONNX Runtime file from $onnxFilePath with ${session.numInputs} inputs and ${session.numOutputs}")
+
+        val aggregatedMap = buildStaticTensors(offenderAndOffences, env) +
+          buildDynamicTensors(offenderAndOffences, env) +
+          buildPreviousOffencesTensors(offenderAndOffences, env) +
+          buildCurrentOffencesTensors(offenderAndOffences, env)
+
+        // Identify if any ONNX input parameters are missing and log an error
+        val missingTensors = session.inputNames.asSequence().minus(aggregatedMap.keys).toList()
```

Comment:
This should perhaps raise an exception once we start using the ONNX version in production